src/Runtime/OMResize.inc

```diff
@@ -0,0 +1,353 @@
+
+  int64_t rank = omTensorGetRank(data);
+  int64_t *inputShape = omTensorGetShape(data);
+  float *scale_factor = NULL;
+  int64_t *output_size = NULL;
+  if (scale_factor_OMT != NULL)
+    scale_factor = (float *)omTensorGetDataPtr(scale_factor_OMT);
+  if (output_size_OMT != NULL)
+    output_size = (int64_t *)omTensorGetDataPtr(output_size_OMT);
+  if (scale_factor == NULL) {
+    scale_factor = (float *)malloc(sizeof(float) * rank);
+    for (int i = 0; i < rank; i++) {
+      scale_factor[i] = ((float)output_size[i]) / inputShape[i];
+    }
+  } else {
+    output_size = (int64_t *)malloc(sizeof(int64_t) * rank);
+    for (int i = 0; i < rank; i++) {
+      output_size[i] = scale_factor[i] * inputShape[i];
+    }
+  }
+
+  int64_t outputSize = 1;
+  for (int i = 0; i < rank; i++) {
+    outputSize *= output_size[i];
+  }
+  float *outputData = (float *)omTensorGetDataPtr(output_OMT);
+
+  // int64_t allCoordinates[outputSize][rank];
+  int64_t *allCoordinates =
+      (int64_t *)malloc(outputSize * rank * sizeof(int64_t));
+  generate_coordinates(rank, output_size, allCoordinates);
+
+  // float coeffs_buffer[coeffs_n]; // = {1.0, 0.};
+  float *coeffs_buffer = (float *)malloc(sizeof(float) * coeffs_n);
+
+  for (int i = 0; i < outputSize; i++) {
+    float *Xs = (float *)malloc(sizeof(float) * rank);
+    for (int j = 0; j < rank; j++) {
+      Xs[j] = *(allCoordinates + i * rank + j);
+    }
+    float r = interpolate_nd_with_x(
+        /*OMTensor */ data,
+        /* n */ 4,
+        /*float scale_factor*/ scale_factor,
+        /*floats *x*/ Xs,
+        /* Coeff_Func_t*/ get_coeffs,
+        /*float */ coeffs_buffer,
+        /*int coeffs_n*/ coeffs_n,
+        /*float roi*/ 0.,
+        /*float extrapolation_value*/ 0.,
+        /*int coordinate_transformation_mode*/ 0,
+        /*exclude */ 0,
+        /*mode */ 0);
+    outputData[i] = r;
+    free(Xs);
+  }
+  if (output_size_OMT == NULL)
+    free(output_size);
+  if (scale_factor_OMT == NULL)
+    free(scale_factor);
+  free(allCoordinates);
+  free(coeffs_buffer);
+}
+
+void Resize_Scales(
+    OMTensor *output, OMTensor *data, OMTensor *scales, char *mode_str) {
+  Coeff_Func_t coeffs_f = NULL;
+  int coeffs_n = 0;
+  if (strcmp(mode_str, "nearest") == 0) {
+    coeffs_f = nearest_coeffs;
+    coeffs_n = 2;
+  } else if (strncmp(mode_str, "linear", 6) == 0) {
+    coeffs_f = linear_coeffs;
+    coeffs_n = 2;
+  } else if (strcmp(mode_str, "cubic") == 0) {
+    coeffs_f = cubic_coeffs;
+    coeffs_n = 4;
+  } else {
+    assert(0 && "Resize runtime: unsupported mode");
+  }
+  interpolate_nd_OMTensor(
+      /*OMTensor */ output,
+      /*OMTensor */ data,
+      /*mode*/ 0,
+      /*OMTensor output size */ NULL,
+      /*OMTensor scales */ scales,
+      /* Coeff_Func_t*/ coeffs_f,
+      /*int coeffs_n*/ coeffs_n,
+      /* roi */ NULL,
+      /*float * extrapolation_value*/ 0,
+      /*int coordinate_transformation_mode*/ 0,
+      /*exclude */ 0);
+}
+
+void Resize_Size(
```

Comment:
[nit]: I would name this `resize_size` for consistency. Similar for `resize_scales`.

Reply:
Resize comes from the ONNX Op name and Size or Scales comes from the attribute name. I can change function name for these two. But I am thinking of constructing the function name automatically.  So can we just leave them?